File: client/src/app/entities/request/create-user-request.ts
export class CreateUserRequest {

    public userName: string
    public displayName: string
    public password: string

    constructor(init?: CreateUserRequest) {
        Object.assign(this, init)
    }

    getUserName() {
        return this.userName
    }

    getPassword() {
        return this.password
    }
}

File: api/src/main/java/com/example/user/controller/dto/request/CreateUserRequest.java
package com.example.user.controller.dto.request;

import com.example.user.controller.dto.annotation.UniqueUserName;
import io.micronaut.serde.annotation.Serdeable;
import jakarta.validation.constraints.NotBlank;

@Serdeable
public record CreateUserRequest(@UniqueUserName String userName,
                                @NotBlank String displayName,
                                @NotBlank String password) {

}